### PR TITLE
dev-db/mariadb: add 10.6.19, 10.11.9, 11.4.3

### PR DESCRIPTION
**Pull request by hydrapolic**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by hydrapolic** [comment]
Hello @arkamar

**Comment by anthonyryan1** [comment]
Looks like each release series has bumped a point release since this was opened. Probably worth updating for those. Then hopefully we can get some tree attention.

File: dev-db/mariadb/mariadb-11.4.3.ebuild [hunk 1]
```diff
@@ -281,7 +281,7 @@ src_configure() {
 	# bug #855233 (MDEV-11914, MDEV-25633) at least
 	filter-lto
 	# bug 508724 mariadb cannot use ld.gold
-	tc-ld-disable-gold
+	tc-ld-force-bfd
```
Comment (arkamar) [inline dev-db/mariadb/mariadb-11.4.3.ebuild:284]: What is the reason for this change? I miss some description anywhere, e.g. in the commit message and/or in the comment above this line.

File: dev-db/mariadb/mariadb-11.4.3.ebuild [hunk 2]
```diff
@@ -9,11 +9,15 @@ JAVA_PKG_OPT_USE="jdbc"
 inherit systemd flag-o-matic prefix toolchain-funcs \
 	multiprocessing java-pkg-opt-2 cmake
 
+DESCRIPTION="An enhanced, drop-in replacement for MySQL"
 HOMEPAGE="https://mariadb.org/"
 SRC_URI="mirror://mariadb/${PN}-${PV}/source/${P}.tar.gz
-	https://github.com/hydrapolic/gentoo-dist/raw/main/mariadb/mariadb-11.4.2-patches-01.tar.xz"
+	https://github.com/hydrapolic/gentoo-dist/raw/main/mariadb/${PN}-11.4.3-patches-01.tar.xz"
```
Comment (arkamar) [inline dev-db/mariadb/mariadb-11.4.3.ebuild:15]: Why do we need the `0005-disable-ed25519-for-server.patch`? I miss this information in the patch.
```diff
# partially revert https://github.com/mariadb-corporation/mariadb-connector-c/commit/dddcf400af9a693fdbed4e692d71bf98b79b7aa1
diff --git a/libmariadb/plugins/auth/CMakeLists.txt b/libmariadb/plugins/auth/CMakeLists.txt
index 3d2ede14..c55bb178 100644
--- a/libmariadb/plugins/auth/CMakeLists.txt
+++ b/libmariadb/plugins/auth/CMakeLists.txt
@@ -69,11 +69,6 @@ IF(CRYPTO_PLUGIN)
                 INCLUDES ${REF10_DIR}
                 LIBRARIES ${CRYPT_LIBS}
                 COMPILE_OPTIONS -DMYSQL_CLIENT=1)
-  SET_PROPERTY(TARGET client_ed25519 PROPERTY C_STANDARD 11)
-
-  IF(HAVE_C_THREAD_LOCAL)
-    TARGET_COMPILE_DEFINITIONS(client_ed25519 PRIVATE -DHAVE_THREAD_LOCAL)
-  ENDIF()

   IF(MSVC)
      # Silence conversion (integer truncation) warnings from reference code
```

**Comment by arkamar** [comment]
My tests finished with following failures:
```
mariadb-10.11.9/temp/build.log:9693:Failing test(s): main.mysqld--help-aria main.log_slow
mariadb-10.6.19/temp/build.log:9432:Failing test(s): main.mysqld--help-aria main.log_slow
mariadb-11.4.3/temp/build.log:9952:Failing test(s): main.ssl_autoverify main.mysqld--help-aria sysschema.v_privileges_by_table_by_level main.log_slow
```

File: dev-db/mariadb/mariadb-11.4.3.ebuild (hunk 1 above)
Comment (hydrapolic) [reply dev-db/mariadb/mariadb-11.4.3.ebuild:284]: It's from `pkgcheck`:

```
DeprecatedEclassFunction: version 10.11.9: uses deprecated function on line 284: tc-ld-disable-gold (migrate to tc-ld-force-bfd)
```

File: dev-db/mariadb/mariadb-11.4.3.ebuild (hunk 2 above)
Comment (hydrapolic) [reply dev-db/mariadb/mariadb-11.4.3.ebuild:15]: The configuration stage failed with:
```
...
-- Looking for floor
-- Looking for floor - not found
-- Looking for pthread_getspecific
-- Looking for pthread_getspecific - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- TLS library/version: OpenSSL 3.3.2
-- SYSTEM_LIBS /usr/lib64/libz.so;dl;m;dl;m;/usr/lib64/libssl.so;/usr/lib64/libcrypto.so
CMake Error at libmariadb/plugins/auth/CMakeLists.txt:72 (SET_PROPERTY):
  SET_PROPERTY could not find TARGET client_ed25519.  Perhaps it has not yet
  been created.
Call Stack (most recent call first):
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:435 (INCLUDE)


CMake Error at libmariadb/plugins/auth/CMakeLists.txt:75 (TARGET_COMPILE_DEFINITIONS):
  Cannot specify compile definitions for target "client_ed25519" which is not
  built by this project.
Call Stack (most recent call first):
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:435 (INCLUDE)
```

**Comment by hydrapolic** [comment]
> My tests finished with following failures:
> 
> ```
> mariadb-10.11.9/temp/build.log:9693:Failing test(s): main.mysqld--help-aria main.log_slow
> mariadb-10.6.19/temp/build.log:9432:Failing test(s): main.mysqld--help-aria main.log_slow
> mariadb-11.4.3/temp/build.log:9952:Failing test(s): main.ssl_autoverify main.mysqld--help-aria sysschema.v_privileges_by_table_by_level main.log_slow
> ```

```
main.log_slow                            w1 [ fail ]
        Test ended at 2024-11-12 12:18:08

CURRENT_TEST: main.log_slow
--- /var/tmp/portage/dev-db/mariadb-10.11.9/work/mysql/mysql-test/main/log_slow.result  2024-08-03 07:29:57.000000000 +0000
+++ /var/tmp/portage/dev-db/mariadb-10.11.9/work/mysql/mysql-test/main/log_slow.reject  2024-11-12 12:18:08.635055199 +0000
@@ -200,7 +200,7 @@
 b`;
 select count(*) from mysql.global_priv where length(priv)>2;
 count(*)
-5
+4
 drop database `a
 b`;
 use test;

Result content mismatch
```

**Comment by arkamar** [comment]
I have found following issue related to the `mail.log_slow` failure: https://jira.mariadb.org/browse/MDEV-27044. I am not sure if it is related, because we didn't hit the failure in 10.6.18, but the issue is quite old.

**Comment by arkamar** [comment]
my local `10.6.20` version bump fails the same.

**Comment by arkamar** [comment]
I pushed `10.6.20` and `10.11.10`

File: dev-db/mariadb/mariadb-11.4.3.ebuild (hunk 2 above)
Comment (arkamar) [reply dev-db/mariadb/mariadb-11.4.3.ebuild:15]: This might be caused by a wrongly updated patch, see the diff between original and fixed patch:
```diff
 diff --git a/libmariadb/libmariadb/CMakeLists.txt b/libmariadb/libmariadb/CMakeLists.txt
-index 852be8dc..0373eb4c 100644
+index 1985b44f..5b77ec40 100644
 --- a/libmariadb/libmariadb/CMakeLists.txt
 +++ b/libmariadb/libmariadb/CMakeLists.txt
-@@ -447,8 +447,10 @@ IF(WIN32)
+@@ -449,8 +449,10 @@ IF(WIN32)
                     "FILE_DESCRIPTION:Dynamic lib for client/server communication")
  ENDIF()

 -ADD_LIBRARY(mariadbclient STATIC  ${MARIADB_OBJECTS} ${EMPTY_FILE})
--TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS} ${CRYPTO_LIBS})
+-TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS} ${CRYPT_LIBS})
 +IF(ENABLE_STATIC_LIBS OR NOT WITHOUT_TOOLS)
 + ADD_LIBRARY(mariadbclient STATIC  ${MARIADB_OBJECTS} ${EMPTY_FILE})
-+ TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS} ${CRYPTO_LIBS})
++ TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS} ${CRYPT_LIBS})
 +ENDIF(ENABLE_STATIC_LIBS OR NOT WITHOUT_TOOLS)

  IF(UNIX)
    ADD_LIBRARY(libmariadb SHARED ${libmariadb_RC} ${MARIADB_OBJECTS} ${EMPTY_FILE})
```
I am running tests for `11.4.4` locally.

Comment (hydrapolic) [reply dev-db/mariadb/mariadb-11.4.3.ebuild:15]: Oops, sorry for that!

**Comment by arkamar** [comment]
I pushed `11.4.4`, how about this PR now? Should I commit the old versions clean up?

**Comment by hydrapolic** [comment]
> I pushed `11.4.4`, how about this PR now? Should I commit the old versions clean up?

Huge thanks @arkamar! Yes please, let's do the cleanup.

**Comment by arkamar** [comment]
> Yes please, let's do the cleanup.

I cherry-picked the cleanup commit and closed this PR with it. Throwing the `dev-db/mariadb: add 10.6.19, 10.11.9, 11.4.3` commit away.
